server/src/main/java/io/crate/planner/WhereClauseOptimizer.java

```diff
@@ -212,16 +215,10 @@ private static List<ColumnIdent> pkColsInclVersioning(DocTableInfo table,
                                                           boolean versionInQuery,
                                                           boolean seqNoAndPrimaryTermInQuery) {
         if (versionInQuery) {
-            ArrayList<ColumnIdent> pkCols = new ArrayList<>(table.primaryKey().size() + 1);
-            pkCols.addAll(table.primaryKey());
-            pkCols.add(DocSysColumns.VERSION);
-            return pkCols;
+            return Collections.unmodifiableList(Lists2.concat(table.primaryKey(), DocSysColumns.VERSION));
         } else if (seqNoAndPrimaryTermInQuery) {
-            ArrayList<ColumnIdent> pkCols = new ArrayList<>(table.primaryKey().size() + 1);
-            pkCols.addAll(table.primaryKey());
-            pkCols.add(DocSysColumns.SEQ_NO);
-            pkCols.add(DocSysColumns.PRIMARY_TERM);
-            return pkCols;
+            Collections.unmodifiableList(
+                Lists2.concat(table.primaryKey(), List.of(DocSysColumns.SEQ_NO, DocSysColumns.PRIMARY_TERM)));
```

Comment:
Why the change here? I think the old version is more efficient as it avoids an intermediate list creation

Reply:
You're right, was not intending to keep that, because of the `List.of()`.